src/components/stateTracker/js/StateTracker.js

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_2_0_0 = fluid_2_0_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    fluid.registerNamespace("fluid.stateTracker");
+
+    fluid.defaults("fluid.stateTracker", {
+        gradeNames: ["fluid.component"],
+        events: {
+            onStateChange: null
+        },
+        members: {
+            // default polling frequency in msec.
+            interval: 10
+        },
+        invokers: {
+            startTracking: {
+                funcName: "fluid.stateTracker.startTracking",
+                args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                                 // changeEvaluator, changeListener, polling interval
+            },
+            stopTracking: {
+                funcName: "fluid.stateTracker.stopTracking",
+                args: ["{that}", "{arguments}.0", "{arguments}.1"]
+                                 // changeListener, intervalID
+            },
+            monitorChange: {
+                funcName: "fluid.stateTracker.monitorChange",
+                args: ["{that}", "{arguments}.0"]
+                                 // monitor info structure
+            },
+            initMonitorInfo: {
+                funcName: "fluid.stateTracker.initMonitorInfo",
+                args: ["{arguments}.0"]
+                       // changeEvaluator object or function
+            }
+        }
+    });
+
+    // Initiate polling.
+    // @param changeEvaluator - an object used to evaluate if the state
+    //                          has changed.  It must have an evaluateChange()
+    //                          function.
+    // @param changeListener  - a listener to handle the onStateChange event.
+    // @param interval        - optional delay between calls to check the state's
+    //                          current value (msec).
+    // @return                - the intervalID.
+    fluid.stateTracker.startTracking = function (that, changeEvaluator, changeListener, interval) {
+        var monitor = that.initMonitorInfo(changeEvaluator);
+        that.events.onStateChange.addListener(changeListener);
+        if (!!interval) {
```

Comment:
For every JavaScript value which is truthy, !! of the value is also truthy, and similarly for falsy values, so this double negation isn't necessary - https://developer.mozilla.org/en-US/docs/Glossary/Truthy